src/testing/builder/CategoriesMapModelBuilder.js

```diff
@@ -28,50 +31,55 @@ class CategoriesMapModelBuilder {
     return {
       [url]: {
         filePath: `path/to/documentDir/resource-cache/v1/some-city/files/${hash}.png`,
-        lastUpdate: moment(lastUpdate).add(this.predictableNumber(index), 'days'),
+        lastUpdate: moment(lastUpdate).add(this._predictableNumber(index), 'days'),
         hash
       }
     }
   }
 
-  addChildren (
+  _addChildren (
     category: CategoryModel,
+    id: number,
     categories: Array<CategoryModel>,
     resourceCache: LanguageResourceCacheStateType,
-    depth: number) {
+    depth: number,
+    ancestors: Array<number>) {
+    categories.push(category)
+    ancestors.push(id)
+
     if (depth === this._depth) {
       return
     }
 
-    for (let i = 0; i < this._maxWidth; i++) {
-      const path = `${category.path}/category_${depth}-${i}`
+    for (let i = 0; i < this._arity; i++) {
+      const path = `${category.path}/category_${i}`
       const lastUpdate = moment('2017-11-18T19:30:00.000Z', moment.ISO_8601)
+      const id = depth * this._arity + i + 1
+      const resourceUrl1 = `https://integreat/title_${depth}-${i}-300x300.png`
+      const resourceUrl2 = `https://integreat/category_${depth}-${i}-300x300.png`
```

Comment:
The resourceUrls are still not unique. (Don't know if that's the expected behavior)

Reply:
This is not yet done.

Reply:
I switched to an other approach for generating unique resource urls

Reply:
Looking good :+1: